components/Common/Pagination/index.tsx

```diff
@@ -0,0 +1,81 @@
+'use client';
+import { ArrowRightIcon, ArrowLeftIcon } from '@heroicons/react/20/solid';
+import type { FC } from 'react';
+import { FormattedMessage, useIntl } from 'react-intl';
+
+import Button from '@/components/Common/Button';
+
+import styles from './index.module.css';
+import { useGetPageElements } from './useGetPageElements';
+
+type Page = {
+  url: string;
+};
+
+export type PaginationProps = {
+  /**
+   * One-based number of the current page
+   */
+  currentPage: number;
+  pages: Page[];
+  /**
+   * The number of page buttons on each side of the current page button
+   * @default 1
+   */
+  currentPageSiblingsCount?: number;
+};
+
+const Pagination: FC<PaginationProps> = ({
+  currentPage,
+  pages,
+  currentPageSiblingsCount = 1,
+}) => {
+  const intl = useIntl();
+
+  const parsedPages = useGetPageElements(
+    currentPage,
+    pages,
+    currentPageSiblingsCount
+  );
+
+  return (
```

Comment:
I wonder how accessibility comes in play here. Can you navigate this Pagination just with ... keyboard strokes?

Reply:
You can navigate it with tabs. There's no strong guidelines nor an official WCAG 2.1 recipe for Pagination components. However, there's tons of resources mentioning some of the core characteristics they can have:

- It should wrap everything through a nav or div with role="navigation"
- The pagination buttons should be wrapped in an ordered list, to let screen readers know the elements are ordered
- The list items should use `aria-setsize` and `aria-posinset`, as most of the times not all list items will be shown at once, so as to let screen readers know what the entire list size is, and the position of each list item in such list
- The list items should contain anchor links
- The attribute aria-current="page" should be used to indicate that such anchor link represents the current page

React MUI's [pagination component](https://mui.com/material-ui/react-pagination/) works with tabs too.

Reply:
Right, you did your due diligence. This looks good, and I'm satisfied with your explanation :)

Reply:
Thank you ✨